zcash_client_backend/src/proto.rs

```diff
@@ -311,46 +341,41 @@ impl proposal::Proposal {
     ) -> Option<Self> {
         let transaction_request = value.transaction_request().to_uri(params)?;
 
-        let transparent_inputs = value
+        let anchor_height = value
+            .shielded_inputs()
+            .map_or_else(|| 0, |i| u32::from(i.anchor_height()));
+
+        let inputs = value
             .transparent_inputs()
             .iter()
             .map(|utxo| proposal::ProposedInput {
                 txid: utxo.outpoint().hash().to_vec(),
+                value_pool: proposal::ValuePool::Transparent.into(),
                 index: utxo.outpoint().n(),
                 value: utxo.txout().value.into(),
             })
+            .chain(value.shielded_inputs().iter().flat_map(|s_in| {
+                s_in.notes().iter().map(|rec_note| proposal::ProposedInput {
+                    txid: rec_note.txid().as_ref().to_vec(),
+                    value_pool: proposal::ValuePool::Sapling.into(),
```

Comment:
Bug: `s_in.notes()` returns a non-empty iterator of notes that can be from Sapling and/or Orchard. `rec_note.note()` needs to be mapped from `WalletNote` (or `ShieldedNote` / `Note` per my other renaming recommendation) to `ValuePool`.

Reply:
It's not yet a bug because we don't yet have shielded notes. But this can be fixed now, I guess, so that it doesn't get missed later.